pom.xml

```diff
@@ -8,28 +8,28 @@
 		<version>2.3.5.RELEASE</version>
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
-	<groupId>io.sentry.demos</groupId>
+	<groupId>io.sentry.samples</groupId>
 	<artifactId>example</artifactId>
 	<version>0.0.1-SNAPSHOT</version>
 	<name>example</name>
 	<description>Demo project for Sentry in Java Spring Boot with Logback</description>
 
 	<properties>
-		<java.version>11</java.version>
+		<java.version>1.8</java.version>
 	</properties>
 
 	<dependencies>
 		<!-- Sentry library for Logback -->
 		<dependency>
 			<groupId>io.sentry</groupId>
 			<artifactId>sentry-logback</artifactId>
-			<version>3.1.1</version>
+			<version>4.1.0</version>
 		</dependency>
 		<!-- Sentry SDK for Java Spring Boot-->
 		<dependency>
 			<groupId>io.sentry</groupId>
 			<artifactId>sentry-spring-boot-starter</artifactId>
-			<version>3.1.1</version>
+			<version>4.1.0</version>
```

Comment:
I'd recommend looking into 4.2.0 though
https://github.com/getsentry/sentry-java/releases/tag/4.2.0
https://docs.sentry.io/platforms/java/migration/#migrating-from-iosentrysentry-410-to-iosentrysentry-420

a lot of bug fixes & improvements

Reply:
@marandaneto the `throw new RuntimeException(message);` in the /checkout endpoint does not get captured by sentry if I use 4.2.0.

I think i'm going to use 4.1.0 for now.

see discuss-sdk's for more detail.

Reply:
I checked the Migration guide, I didn't find anything that i need to do. I saw stuff about Transactions and Handled errors, but this is an unhandled error and i'm not doing anything with transactions.

weird that the /unhandled endpoint does the same thing `throw new RuntimeException` and works fine.

Reply:
mmm, I think it could be related to https://docs.sentry.io/platforms/java/migration/#sentryexceptionresolver-does-not-send-handled-errors-by-default
@bruno-garcia already told @maciejwalkowiak let's see what he finds out

Reply:
@thinkocapo `throw new RuntimeException` is later on caught and rethrown as `ResponseStatusException` - which means that the exception is handled and you intentionally set what status code should be returned to the client. It's not considered as an unhandled exception and since 4.2.0 we don't log it to Sentry by default.

Reply:
thanks @Maciejwalkowiak @marandaneto I missed that. I assumed it was unhandled because that's how our other demo's are.

Can you confirm, why is this error handled error and captured by Sentry? Is this because [SentryExceptionResolver and @ExceptionHandlers](https://docs.sentry.io/platforms/java/migration/#sentryexceptionresolver-does-not-send-handled-errors-by-default-1
) are not catching it?
```
	@GetMapping("/handled-and-captured-by-sentry")
	public String HandledAndCapturedBySentryError() {
		String someLocalVariable = "stack locals";

		throw new RuntimeException("Handled and captured by sentry!");
	} 
```


Trying to understand why this handled exception is _not_ captured by sentry:
```
throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
```

But this handled exception is captured by sentry:
```
throw new RuntimeException("Handled and captured by sentry!");
```

Any idea?

Checking that it is handled...
![image](https://user-images.githubusercontent.com/8920574/109508284-b1ec2800-7a6d-11eb-8e80-4c0342addfe9.png)

Code is currently on 4.2.0

Reply:
`ResponseStatusException` has a special treatment in Spring's `ResponseStatusExceptionResolver` - this is why it does not get reported to Sentry while other exceptions are. Look into the `ResponseStatusExceptionResolver` [sources](https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/mvc/annotation/ResponseStatusExceptionResolver.java).

Reply:
@maciejwalkowiak is there a way to make it get captured by Sentry?
I'm trying to make sense of the recommendations here:
https://docs.sentry.io/platforms/java/migration/#sentryexceptionresolver-does-not-send-handled-errors-by-default

Is this the solution?
```
@EnableSentry(exceptionResolverOrder = -2147483648)
class CustomConfiguration {
  ...
}
```

Maybe something like this?
```
Sentry.captureException(throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());)
```
must use Sentry.captureException?

@marandaneto

Reply:
I looked at [sources](sources) but not sure what to make of this. It must explain why ResponseStatusException does not get captured. Is there something in here I can take advantage of, to make the `ResponseStatusException` get captured by Sentry?

Reply:
Since this is a Spring Boot application - not Spring - everything should be configured in `application.properties` - not using `@EnableSentry`. In this case, to send `ResponseStatusException` to Sentry you have to add following property to `application.properties`:
```
sentry.exception-resolver-order=-1
```
The number can be anything below `0`, as the built in into Spring exception resolver that handles this exception has order `0`.

Reply:
Thanks this fixed it. I would have tried this sooner but I was confused on the docs. I kept thinking the docs were saying, "If your'e getting too many errors, then do xyz" but I kept thinking, "I'm not getting any handled errors, so I don't think this documentation is related to my use case" but I think it was just a bad reading + interpretation on my part, of the documentation. I'm not as familiar with java and its frameworks.

Personally, instead of 'SentryExceptionResolver...' I'd first write something like, "The Sentry SDK will not capture errors if Spring/SpringBoot is already handling them. To fix this, change sentry.exception-resolver-order to a negative value like sentry.exception-resolver-order=-1". it's a clearer call to action.

but it's good to see @ExceptionHandlers written in the docs. I wasn't really sure what SentryExceptionResolver meant, I see now it means something capturing the error, it's part of the Sentry SDK.

Reply:
I think you can make a PR to docs to make it sounds better than it is right now 🙂